include/cctz/civil_time_detail.h

```diff
@@ -91,6 +91,15 @@ CONSTEXPR_F int days_per_4years(int yi) noexcept {
 CONSTEXPR_F int days_per_year(year_t y, month_t m) noexcept {
   return is_leap_year(y + (m > 2)) ? 366 : 365;
 }
+// The compiler cannot optimize away the check if we use
+// fsanitize=array-bounds.
+// When this function is inlined, m is guaranteed to be in [1, 13).
```

Comment:
```
// When this function is inlined, ...
```
Is inlining germane to the issue?

Reply:
yes. just looking at the function without inlining, it is clear that we don't know anything about `m`. but even when we inline, and the compiler should know be able to optimize, it doesn't.

Reply:
reworded the comment.